### DISCUSSION TITLE
Reviews only accepted for orhun

### DISCUSSION THREAD

**Discussion by tessus** [orhun/rustypaste-cli#119] [General]
When I approve a dependabot PR, it is not merged, even if all pipelines pass.

Also, the PRs should still be able to be merged if the beta and nightly fail. Even though I have push access, I cannot merge any dependabot PRs.

@orhun any idea why that is?

**Comment by tessus** [comment]
I suspect that there are certain rules in place tbat prevent me from reviewing and merging.

**Reply by orhun** [reply]
There is a branch protection:

![image](https://github.com/orhun/rustypaste-cli/assets/24392180/50b26649-0db7-4f7d-bc09-58b5cd6b7fbc)

I think you need to add yourself to the CODEOWNERS file.

**Reply by tessus** [reply]
Hmm, interesting. I guess that rule is different than the one on the rustypaste repo. There I didn't have to be a code owner. Additionally I suspect that dependabot must be an exception in the repos where the auto-merge works.

I think when gh came up with the new rulesets some repos have a mix of classic branch protection rules and new rulesets. Unfortunately I cannot help you there since I don't have access to the settings. But at one point you will have to compare your rulesets with a repo where the auto-merge works.

There are also bypass settings and settings for deployments/checks for mergability. e.g. that the beta/nightly tests do not have to succeed to allow a merge. The beta/nightly tests are good to know, but they should not be a reason for a PR not being able to be merged.

**Reply by orhun** [reply]
Yes, one day I should really sit down and figure out everything...

**Reply by tessus** [reply]
There is a status check setting:

<img width="839" alt="image" src="https://github.com/orhun/rustypaste-cli/assets/223439/934b7526-a616-41e5-916b-890dc6493f77">

There you pick all except the beta/nightly.... That should do it...

**Reply by orhun** [reply]
Hmm, that settings is not enabled.

**Reply by tessus** [reply]
maybe then I can just not merge because I am not a CodeOwner and it has nothing to do with the pipeline

**Reply by tessus** [reply]
I can open a PR to add me to the CODEOWNERS file. The question is what are you using for all your other repos. e.g. I didn't have to do that for the rustypaste repo. Maybe you want to keep that consistent. I am ok with it either way. Personally it would be very confusing for me if I had different "standard rules" for all my repos.

**Reply by orhun** [reply]
We can use the CODEOWNER file either way, it's good because you will get review requests as well. I usually add that file to my repos.

**Reply by tessus** [reply]
ok. in that case I'll create PRs in both repos later tonight.
